### PR TITLE
Documentation for `check_code_style.sh`

### PR DESCRIPTION
**Pull request by tammoippen**
This PR depends on #95. Further see #38.

It adds the documentation for the script `extras/check_code_style.sh`.

### CODE REVIEW

File: coding_guidelines_c++.md [hunk 1]
```diff
@@ -31,22 +32,17 @@ _Note that older code might not conform to this rules and should be fixed when c
...
 The [clang-format](http://clang.llvm.org/docs/ClangFormat.html) tool is build on the clang compile fronted. It allows to pretty print input files in a configurable manner, and has vim and emacs integration. We supply a [.clang-format](#clang-format-file) configuration file to enforce some parts of the coding style. During the code review process we check that there is no difference between the committed files and the formatted version of the committed files:
 
-    clang-format <committed file> > tmp_out.txt
-    diff <committed file> tmp_out.txt > /dev/null
-    if [ $? -ne 0 ]; then
-      # error
-    fi
 
-Developer can benefit from the tool by formatting their changes before issuing a pull request: `clang-format -i <committed file>`. We require clang-format version 3.6 or later.
+Developer can benefit from the tool by formatting their changes before issuing a pull request: `clang-format -i <committed file>`. We require clang-format version 3.6.
```
Comment (heplesser) [inline coding_guidelines_c++.md]: Why precisely 3.6? That makes it a bit tricky to keep vera++ locally, as one needs to suppress updates arriving via brew or ports.

File: coding_guidelines_c++.md [hunk 2]
```diff
@@ -78,7 +72,7 @@ OS X:
 
 Or consider installing `vera++` from the [binary packages](https://bitbucket.org/verateam/vera/wiki/Installation).
 
-Finally, we let [cppcheck](http://cppcheck.sourceforge.net/) statically analyse the committed files and check for severe errors:
+Finally, we let [cppcheck](http://cppcheck.sourceforge.net/) statically analyse the committed files and check for severe errors. We require cppcheck version 1.69.
```
Comment (heplesser) [inline coding_guidelines_c++.md]: Same here, why not 1.69 or later?

File: coding_guidelines_c++.md (hunk 1 above)
Comment (tammoippen) [inline coding_guidelines_c++.md]: There a formatting differences between clang-format 3.7 and 3.6. In TravisCI we use 3.6, so the local test should use the same, to reduce surprises.

File: coding_guidelines_c++.md (hunk 2 above)
Comment (tammoippen) [inline coding_guidelines_c++.md]: I guess, we can add 'or later' to cppcheck, as the results here are not critical for success of TravisCI tasks.

File: coding_guidelines_c++.md (hunk 1 above)
Comment (heplesser) [inline coding_guidelines_c++.md]: @tammoippen I understand, but I think this is a rather unfortunate choice by the clang-format developers. The format that you get should be definable independent of the version of the formatter program. Won't we be locked to 3.6 for all eternity now?

- Could you add a short note to the documentation here why we require 3.6?
- Also, could you add information on https://nest.github.io/nest-simulator/coding_guidelines_c++ on how to install a particular version of `clang-format` with brew, and maybe also add the info on why we need 3.6?

Comment (tammoippen) [inline coding_guidelines_c++.md]: Homebrew provides clang-format 3.7 only right now and no recipes for older versions. I supplied a download link of pre-build binaries for clang-format-3.6 for OS X.

**Comment by heplesser** [comment]
:+1: for merging. 

@tammoippen How do you install the pre-built clang-format? I just extracted the archive and added it to my `PATH`.  Any better idea?

**Comment by tammoippen** [comment]
@heplesser this is exactly how i did it.

File: coding_guidelines_c++.md [hunk 3]
```diff
@@ -93,6 +87,49 @@ OS X:
...
 
+Build from scratch:
+
+    git clone https://github.com/danmar/cppcheck.git
+    cd cppcheck
+    git checkout tags/1.69
+    make PREFIX=$PWD/install/ CFGDIR=$PWD/install/cfg HAVE_RULES=yes install
+
+    # In `.bashrc` at install/bin to `PATH`:
+    export PATH=$PATH:/path/to/cppcheck/install/bin
```
Comment (jougs) [inline coding_guidelines_c++.md]: I'd rather use
```
export PATH=/path/to/cppcheck/install/bin:$PATH
```
That way, the local version would even be picked up if a global version was lying around somewhere.

**Comment by tammoippen** [comment]
@jougs @heplesser Ping. :smile:

**Comment by jougs** [comment]
The new version looks good to me. Many thanks.

**Pull request merged**